### PR TITLE
chore: support redocly registry at .com domain

### PR DESCRIPTION
**Pull request by RomanHotsiy**
## What/Why/How?

Some small updates required to migrate registry URLs from api.redoc.ly to api.redocly.com.

## Reference

Depends on https://github.com/Redocly/workflows/pull/1820

## Check yourself

- [x] Code is linted
- [ ] Tested with redoc/reference-docs/workflows
- [x] All new/updated code is covered with tests

## Security

- [x] Security impact of change has been considered
- [x] Code follows company security practices and guidelines

### CODE REVIEW

**Review by slavikbez — APPROVED** [review]

File: packages/core/src/config/load.ts [hunk 1]
```diff
@@ -41,7 +41,7 @@ export async function loadConfig(configPath?: string, customExtends?: string[]):
       },
       //support redocly.com domain for future compatibility
```
Comment (AyeOne) [inline packages/core/src/config/load.ts:42]: It seems that the comment need to be updated.
~Do I understand it correctly, that we don't need similar config for 'eu' region?~
(ok, I see, EU domain is already `.com`)

File: packages/core/src/redocly/index.ts [hunk 2]
```diff
@@ -160,16 +162,15 @@ export class RedoclyClient {
 
 export function isRedoclyRegistryURL(link: string): boolean {
   const domain = process.env.REDOCLY_DOMAIN || DOMAINS[DEFAULT_REGION];
-  if (!link.startsWith(`https://api.${domain}/registry/`)) return false;
-  const registryPath = link.replace(`https://api.${domain}/registry/`, '');
 
-  const pathParts = registryPath.split('/');
+  const legacyDomain = domain === 'redocly.com' ? 'redoc.ly' : domain;
```
Comment (AyeOne) [inline packages/core/src/redocly/index.ts:166]: IMHO, it's a bit hard to understand logic here.

**Review by AyeOne — APPROVED** [review]

File: packages/core/src/redocly/index.ts (hunk 2 above)
Comment (ozapotichnyi) [reply packages/core/src/redocly/index.ts:166]: I think it'll be more simple if you just hardcode 'redoc.ly' (you're comparing it with hardcoded values btw)

**Review by ozapotichnyi — APPROVED** [review]

File: packages/core/src/redocly/index.ts (hunk 2 above)
Comment (RomanHotsiy) [reply packages/core/src/redocly/index.ts:166]: >  if you just hardcode 'redoc.ly' 

In such case someone on EU region can get `redoc.ly/*` links matching too but it should not.

**Review by Marshevskyy — APPROVED** [review]

**Review by tatomyr — APPROVED** [review]

**Pull request merged**